### ISSUE TITLE
Visual Studio 2019 freezes after installing AWS Toolkit extension

### ISSUE THREAD

**Issue by kotha16** [aws/aws-toolkit-visual-studio#93]
**Describe the bug**
Visual Studio 2019 freezes after installing the AWS toolkit for visual studio (2019).

**To Reproduce**
1. Install AWS Toolkit for visual studio 2019 
2. Open Visual studio 2019.  
3. Create a new dotnet project.

**Expected behavior**
Visual Studio should not freeze

**Screenshots**
![aws_toolkit_issue1](https://user-images.githubusercontent.com/9220595/83094885-6d5fd600-a070-11ea-9893-25e6d4d6fae5.PNG)

**Computer (please complete the following information):**

- Windows Version: Windows 10 Enterprise 10.0.17763 Build 17763
- Visual Studio Version: Visual Studio 2019 Enterprise
- AWS Toolkit for Visual Studio Version: 1.17.1.0

**Additional context**

**Comment by awschristou** [comment]
Thanks for reaching out. When Visual Studio is closed, could you please retrieve your log file from `%localappdata%\AWSToolkit\logs\visualstudio` and attach it?

**Comment by kotha16** [comment]
[log.txt](https://github.com/aws/aws-toolkit-visual-studio/files/4696675/log.txt)

Attached is the log file

**Comment by cal5barton** [comment]
This happens all the time. I ended up disabling the extension because it slows everything down when it does work and crashes visual studio most of the time.

**Comment by awschristou** [comment]
Hi @cal5barton I'd be interested in knowing some more details about your environment:

* please attach the toolkit log file from `%localappdata%\AWSToolkit\logs\visualstudio` , particularly if you're encountering visual studio crashes
* if you are loading directly into a solution, how many projects does the solution typically have?
* how many profiles are set up in your AWS credentials file?

**Comment by cal5barton** [comment]
I don't use a credentials file as my AWS account is backed by OKTA so that may be the issue with mine. I do however use a config file which contains 9 different profiles. Looks like in the logs it's crashing when it's trying to load a profile it's wanting MFA options because of OKTA. 

```
2020-08-25 16:40:53,143 [1] WARN  Amazon.AWSToolkit.Navigator.Node.AWSViewModel - Skipping adding profile okta-prod-proxy because getting credentials failed.
Amazon.Runtime.ProcessAWSCredentialException: Command returned non-zero exit value 1 with the error - time="2020-08-25T16:40:53-06:00" level=info msg="Requesting MFA. Please complete two-factor authentication with your second device" 
time="2020-08-25T16:40:53-06:00" level=info msg="Select a MFA from the following list" 
time="2020-08-25T16:40:53-06:00" level=info msg="0: OKTA (push)" 
time="2020-08-25T16:40:53-06:00" level=info msg="1: OKTA (sms)" 
time="2020-08-25T16:40:53-06:00" level=info msg="2: OKTA (token:software:totp)" 
EOF

   at Amazon.Runtime.ProcessAWSCredentials.SetCredentialsRefreshState(ProcessExecutionResult processInfo)
   at Amazon.Runtime.ProcessAWSCredentials.DetermineProcessCredential()
   at Amazon.Runtime.ProcessAWSCredentials.GenerateNewCredentials()
   at Amazon.Runtime.RefreshingAWSCredentials.GetCredentials()
   at Amazon.AWSToolkit.Navigator.Node.AWSViewModel.ProcessStore(List`1 accounts, HashSet`1 alreadyAddedAccessKeys, ICredentialProfileStore store)

```
This happens regardless of if I'm loading into a solution file or with no code.

> Hi @cal5barton I'd be interested in knowing some more details about your environment:
> 
> * please attach the toolkit log file from `%localappdata%\AWSToolkit\logs\visualstudio` , particularly if you're encountering visual studio crashes
> * if you are loading directly into a solution, how many projects does the solution typically have?
> * how many profiles are set up in your AWS credentials file?

**Comment by awschristou** [comment]
Thank you for sharing your setup. When the toolkit starts up, it is trying to check which profiles are valid. It sounds like it is frozen since that profile is waiting on an MFA prompt which will never get fulfilled. Without sharing any of the values, could you please share which keys your profile is using in the config file? I'm interested to know if it is using `credential_process` for example.

We have an open issue requesting SSO support at #98, please take a moment to add a reaction on that issue.

**Comment by cal5barton** [comment]
@awschristou here is an example of two profiles in my config

```
[profile okta-prod-proxy]
credential_process = aws-okta exec okta-prod -- cmd /c C:\batch\proxy-aws-okta-result.bat
region = us-west-2

[profile okta-prod]
output = json
region = us-west-2
aws_saml_url = *****
role_arn = *****
```

**Comment by awschristou** [comment]
Thanks, this has been useful in identifying a scenario where the toolkit negatively affects the toolkit performance. I've written up #115 so that it can be tracked separately.

**Comment by awschristou** [comment]
Version 1.21.0.0 of the Toolkit contains improved credentials support (see #98 for AWS SSO and #46 for MFA). These might help the auth stories mentioned in this issue. Please try it out, and open a new issue if you are still encountering some problems.

Documentation is available at https://docs.aws.amazon.com/toolkit-for-visual-studio/latest/user-guide/credentials.html, and a walk-through of some examples at https://aws.amazon.com/blogs/developer/introducing-aws-toolkit-for-visual-studio-support-for-aws-sso-and-assume-role-with-mfa/
